### DISCUSSION TITLE
The problem of Restart the simulation

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#16538] [Q&A General] [answered]
Dear MOOSE experts, 
Recently, I added the Checkpoint module to the cases [grain_growth_2D_graintracker.i](https://github.com/idaholab/moose/blob/6437ec83fe011ae67c7b71b4210b426db6ca320b/modules/phase_field/examples/grain_growth/grain_growth_2D_graintracker.i) and [poly_grain_growth_2D_eldrforce.i ](https://github.com/idaholab/moose/blob/6437ec83fe011ae67c7b71b4210b426db6ca320b/modules/combined/examples/phase_field-mechanics/poly_grain_growth_2D_eldrforce.i)respectively, and added the code as follows，
```
[Out]
Type = Checkpoint
Interval = 10 # The interval at which time steps are output to the solution file
Num_files = 3 # Number of the Restart files to save
[]
```
For grain_growth_2D_graintracker.i, Restart can be achieved , in other words *-opt -i grain_growth_2D_graintracker.i --reover, can continue the simulation successfully. #16447 

But for poly_grain_growth_2D_eldrforce.i, I first set
start_time = 0,0 end_time = 100
After the simulation is completed, modify it to start_time = 100 and end_time = 200. But it cannot be Restart , the error is displayed as，
```
================================================= =================================
= BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
= PID 587849 RUNNING AT pwmoose-Default-string
= EXIT CODE: 139
= CLEANING UP REMAINING PROCESSES
= YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
================================================= =================================
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Segmentation fault (signal 11)
This typically refers to a problem with your application.
Please see the FAQ page for debugging suggestions
```
So my question is, why do poly_grain_growth_2D_eldrforce.i perform Restart errors, and how should I solve it?
And the information output to the terminal during the simulation process is shown in two  log files.
[poly_grain_growth_2D_eldrforce_init.log](https://github.com/idaholab/moose/files/5715979/poly_grain_growth_2D_eldrforce_init.log)
[poly_grain_growth_2D_eldrforce_recover.log](https://github.com/idaholab/moose/files/5715980/poly_grain_growth_2D_eldrforce_recover.log)

Any suggestions or recommendations to fix these problems would be greatly appreciated.
Thank you
Wei Peng

**Comment by SudiptaBiswas** [comment, marked as answer]
It seems to me you are using `recover` option instead of `restart`. There is a simpler way to restart the simulation. Look into the differences here, https://mooseframework.inl.gov/application_usage/restart_recover.html#!

You can also use the debugger to pinpoint the issue, 
https://mooseframework.inl.gov/application_development/debugging.html#!

**Reply by PengWei97** [reply]
Thank you for your response. I have solved the problem.

**Reply by PengWei97** [reply]
Hi @SudiptaBiswas, maybe this is a very old question, but as of now I still haven't solved it.

I do use `recover`, i.e. the simulation needs to be recovered due to a failure or premature termination during the calculation. 
Here's how I did it, and I desperately need your help,
First, I run `~/projects/luwu/luwu-opt -i poly_grain_growth_2D_eldrforce.i`, then use `ctrl+D` to manually terminate it after four steps of calculation;
Next, I use the command `~/projects/luwu/luwu-opt -i poly_grain_growth_2D_eldrforce.i --recover` to use the `recover` function so that it can continue from the previous step, but the following error occurred,
```bash
*** Warning ***
Mesh unchanged, skipping remaining steps...

Time Step 5, time = 5.20912, dt = 1.45152

Grain Tracker Status:
Grains active index 0: 1 -> 1
Grains active index 1: 1 -> 1
Grains active index 2: 1 -> 1
Grains active index 3: 1 -> 1
Grains active index 4: 1 -> 1

Finished inside of GrainTracker

Segmentation fault (core dumped)
```

Following your suggestion, I tried to use Debugging to find where the problem is, but in Debuggers I don't know how to use the `recover` function, even typing `gdb --args ~/projects/luwu/luwu-dbg -i poly_grain_growth_2D_eldrforce.i --reover `, it also starts from 0.

So my problem now is that while I now know the problem is in the `graintraker`, I don't know how to use the `recover` function in debug mode.

Any suggestions or recommendations to fix these problems would be greatly appreciated.

**Reply by GiudGiud** [reply]
[#22450](https://github.com/idaholab/moose/pull/22450) got merged and should help with this restart

**Reply by eric199405** [reply]
> Thank you for your response. I have solved the problem.

Hi Pengwei,

This is Zixu Guo from NUS. 

May I ask how did you solve such problem, like restart and exit code 139.

Thanks,
